src/Stratis.Features.FederatedPeg/FederatedPegOptions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Features.FederatedPeg
+{
+    public class FederatedPegOptions
+    {
+        public IServiceCollection Services { get; }
```

Comment:
Ouch this feels very wrong....

Reply:
It's quite similar to how .NET Core injection happens.

Maybe have a look at how SmartContractOptions is used in `CirrusD/Progam.cs` to give access to different options (whitelisting, VM type, etc).

Reply:
I also don't like this. 

With the change I suggested above adding services wont be needed.